pkg/parse/run.go

```diff
@@ -368,12 +369,22 @@ func parseAndUpdate(ctx context.Context, p Parser, trigger string, state *reconc
 		lastUpdate: metav1.Now(),
 	}
 	if state.needToSetSyncStatus(newSyncStatus) {
-		if err := p.SetSyncStatus(ctx, syncErrs); err != nil {
+		// Set Syncing=False because the Applier has stopped.
+		if err := p.SetSyncStatus(ctx, false, syncErrs); err != nil {
 			syncErrs = status.Append(syncErrs, err)
 		} else {
 			state.syncStatus = newSyncStatus
 			state.syncingConditionLastUpdate = newSyncStatus.lastUpdate
 		}
+		// Prepend any remediator errors
+		// TODO: combine this with SetSyncStatus so it's a single update
+		remediatorErrs := p.RemediatorConflictErrors()
+		if len(remediatorErrs) > 0 {
+			if err := UpdateConflictManagerStatus(ctx, remediatorErrs, p.K8sClient()); err != nil {
+				syncErrs = status.Append(syncErrs, err)
+			}
+		}
+		// TODO: Am I suppossed to update state.syncStatus or state.syncingConditionLastUpdate again?
```

Comment:
I really dont understand how this works.

Why is it called a `syncStatus` if it's a `sourceStatus`?

Why does it update the `state` on success to a hypothetical state that isn't actually passed into `SetSyncStatus`?

Why does it store a `now()` and then generate a new now timestamp when setting the sync status?

Why does it update the status and then modify the status? This should all be one big reconcile function, not a series of updates that drop and re-add errors....